GitUI/Editor/FileViewer.cs

```diff
@@ -395,6 +398,55 @@ string GetFileText()
             }
         }
 
+        private Task ShowOrDeferAsync(string fileName, Func<Task> showFunc)
+        {
+            return ShowOrDeferAsync(GetFileLength(), showFunc);
+
+            long GetFileLength()
+            {
+                var file = new FileInfo(fileName);
+
+                if (!file.Exists)
+                {
+                    file = new FileInfo(_fullPathResolver.Resolve(fileName));
+                }
+
+                if (file.Exists)
+                {
+                    return file.Length;
+                }
+
+                return 0;
```

Comment:
If file doesn't exist should we return `-1`? This way we can tell zero-length files from absent ones (i.e. skip line 440 completely)...

Reply:
Can do. But this is a local function where the resulting value just has to be less than some threshold (currently 5MB) for the file to show.

Reply:
Added an explanatory comment here. Does that help?